src/mc.rs

```diff
@@ -431,4 +434,73 @@ mod tests {
             }
         }
     }
+
+    mod query_public_addr {
+        use super::*;
+        use bytes::buf::FromBuf;
+
+        #[test]
+        fn when_protocol_is_tcp_it_sends_encrypted_echo_address_request() {
+            let mut evloop = unwrap!(Core::new());
+            let handle = evloop.handle();
+            let listener = unwrap!(TcpListener::bind(&addr!("0.0.0.0:0"), &handle));
+            let server_addr = unwrap!(listener.local_addr()).unspecified_to_localhost();
+            let incoming = listener.incoming();
+
+            let server_info = PeerInfo::with_rand_key(server_addr);
+            let query =
+                query_public_addr(Protocol::Tcp, &addr!("0.0.0.0:0"), &server_info, &handle)
+                    .then(|_| Ok(()));
+
+            let make_query = incoming.into_future()
+                .map_err(|e| panic!(e))
+                .map(|(client_opt, _incoming)| {
+                    client_opt.expect("Failed to receive client connection")
```

Comment:
Generally in our tests we should use `unwrap!(..., "this is a panic message")` rather than `expect`.

Reply:
right, I'll fix it ;)